odpf/entropy/v1beta1/service.proto

```diff
@@ -0,0 +1,100 @@
+syntax = "proto3";
+
+package odpf.entropy.v1beta1;
+
+import "google/api/annotations.proto";
+import "google/protobuf/field_mask.proto";
+import "google/protobuf/struct.proto";
+
+option go_package = "github.com/odpf/proton/entropy/v1beta1;entropyv1beta1";
+option java_multiple_files = true;
+option java_package = "io.odpf.proton.entropy";
+option java_outer_classname = "EntropyServiceProto";
+
+
+service ResourceService {
+  rpc ListResources(ListResourcesRequest) returns (ListResourcesResponse) {
+    option (google.api.http) = {
+      get: "/v1beta1/resources/"
+    };
+  }
+
+  rpc GetResource(GetResourceRequest) returns (GetResourceResponse) {
+    option (google.api.http) = {
+      get: "/v1beta1/resources/{urn}"
+    };
+  }
+
+  rpc CreateResource(CreateResourceRequest) returns (CreateResourceResponse) {
+    option (google.api.http) = {
+      post: "/v1beta1/resources"
+      body: "resource"
+    };
+  }
+
+  rpc UpdateResource(UpdateResourceRequest) returns (UpdateResourceResponse) {
+    option (google.api.http) = {
+      patch: "/v1beta1/resources/{resource.urn}"
+      body: "resource"
+    };
+  }
+
+  rpc DeleteResource(DeleteResourceRequest) returns (DeleteResourceResponse) {
+    option (google.api.http) = {
+      delete: "/v1beta1/resources/{urn}"
+    };
+  }
+}
+
+message Resource {
+  string id = 1;
+  string urn = 2;
+  string name = 3;
+  string parent = 4;
+  string kind = 5;
+  google.protobuf.Value configs = 6;
+  string status = 7;
+  map<string, string> labels = 8;
+}
+
+message ListResourcesRequest {
+  string parent = 1;
+  string kind = 2;
+}
+
+message ListResourcesResponse {
+  repeated Resource resources = 1;
+}
+
+message GetResourceRequest {
+  string urn = 1;
+  string kind = 2;
```

Comment:
Do we see urn as unique? How do we use kind here? And what about id? How is that being utilized?
Same in `Delete` call as well?

Reply:
We plan to get rid of `id` and make `urn` as our unique identifier.

Reply:
Okay, and what about `kind`? Are `urn` not unique enough to find the resource?

Reply:
Yes we will get rid of `kind` as well. At first the responsibility of generating and maintaining `urn` was with each modules package, but we have moved that into a central package allowing us to have proper constraints on it.

Reply:
Fixed this.